README.md

```diff
@@ -120,6 +120,68 @@ If you prefer live discussions, some of us also hang out in
 
 ## FAQ
 
+#### How do I configure alternative storage backends for LXD?
+
+LXD supports various storage backends; below are instructions on how to
+configure some of them. By default, we use a simple directory backed storage
+mechanism, but we recommend using ZFS for best results.
+
+###### ZFS
+
+First, you need to install the ZFS tooling. On Wily and above this is just:
+
+    sudo apt-get install zfsutils-linux
+
+ZFS has many different ways to procure a zpool, which is what you need to feed
+LXD. For example, if you have an extra block device laying around, you can
+just:
+
+    sudo zpool create lxd /dev/sdc6
+
+However, if you want to test things out on a laptop or don't have an extra disk
+laying around, ZFS has its own loopback driver and can be used directly on a
+(sparse) file. To do this, first create the sparse file:
+
+    sudo dd if=/dev/zero of=/var/lib/lxd.img bs=1k count=1 seek=100M
+
+then,
+
+    sudo zpool create lxd /var/lib/lxd.img
+
+Finally, whichever method you used to create your zpool, you need to tell LXD
+to use it:
+
+    lxc config set storage.zfs_pool_name lxd
+
+###### BTRFS
+
+The setup for btrfs is fairly simple, just mount /var/lib/lxd (or whatever your
+chosen `LXD_DIR` is) as a btrfs subvolume before you start LXD, and you're good
+to go. First install the btrfs userspace tools,
+
+    sudo apt-get install btrfs-tools
+
+Now, you need to create a btrfs filesystem. If you don't have an extra disk
+laying around, you'll have to create your own loopback device manually:
+
+    sudo dd if=/dev/zero of=/var/lib/lxd.img bs=1k count=1 seek=100M
+    sudo losetup /dev/loop0 /var/lib/lxd.img
+
+Once you've got a loopback device (or an actual device), you can create the
+btrfs filesystem and mount it:
+
+    sudo mkfs.btrfs /dev/loop0 # or your real device
+    sudo mount /dev/loop0 /var/lib/lxd
+
+###### LVM
+
+To set up LVM, there is a script provided in
+[/scripts/lxd-setup-lvm-storage](https://raw.githubusercontent.com/lxc/lxd/master/scripts/lxd-setup-lvm-storage)
+which will do it for you. It can be run via:
+
+    sudo apt-get install lvm2
+    ./scripts/lxd-setup-lvm-storage -s 10G
+
```

Comment:
On Thu, Nov 05, 2015 at 10:28:17AM -0800, Michael McCracken wrote:
> > +
> > +Once you've got a loopback device (or an actual device), you can create the
> > +btrfs filesystem and mount it:
> > +
> > +    sudo mkfs.btrfs /dev/loop0 # or your real device
> > +    sudo mount /dev/loop0 /var/lib/lxd
> > +
> > +###### LVM
> > +
> > +To set up LVM, there is a script provided in
> > +[/scripts/lxd-setup-lvm-storage](https://raw.githubusercontent.com/lxc/lxd/master/scripts/lxd-setup-lvm-storage)
> > +which will do it for you. It can be run via:
> > +
> > +    sudo apt-get install lvm2
> > +    ./scripts/lxd-setup-lvm-storage -s 10G
> > +
> 
> A couple notes - you also should apt install thin-provisioning-tools.

Ok, the script should probably check for that too then :)

> The script is only for setting up a VG with one loopdev file PV. If you want to use a block device you need this:
> ```
> sudo pvcreate /dev/sdxy
> sudo vgcreate lxd_takes_this_whole_vg /dev/sdxy
> lxc config set storage.lvm_vg_name lxd_takes_this_whole_vg
> ```
> 
> Might also mention that the script has a --destroy arg to clean up all the LVM bits and bobs for you and delete your loopdev image file

Will do.

> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/lxc/lxd/pull/1274/files#r44048764